bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -2952,8 +2952,11 @@ StyleItem[] merge (long items, int itemCount) {
 			linkBefore = false;
 		}
 		char ch = segmentsText.charAt(start);
+		if(ch == '\r') {
+			ch = segmentsText.charAt(start+1);
+		}
 		switch (ch) {
-			case '\r':
+			//case '\r':
```

Comment:
You can remove this line , if commented out

Reply:
We will retain this as people looking at this part of code should get to know that there was earlier implementation which was the cause of this extra line break which is '\r' case and the respective handling of indexes is all about the whole fix proposed in this pr.

Reply:
It would be better to write an explicit comment instead e.g.
```suggestion
			// '\r': is already handled above
```

on the other hand I wonder if the switch is here of any use... e.g. you can write:

item.lineBreak = ch=='\n'
item.tab = ch=='\t'

Reply:
done with the suggestions.